src/Module.ts

```diff
@@ -385,6 +388,41 @@ export default class Module {
 		return makeLegal(ext ? base.slice(0, -ext.length) : base);
 	}
 
+	mark () {
+		this.getExports().forEach(name => {
+			const variable = this.traceExport(name);
+
+			variable.exportName = name;
+			variable.includeVariable();
+
+			if (variable.isNamespace) {
+				(<NamespaceVariable>variable).needsNamespaceBlock = true;
+			}
+		});
+
+		this.getReexports().forEach(name => {
+			const variable = this.traceExport(name);
+
+			if (variable.isExternal) {
+				variable.reexported = (<ExternalVariable>variable).module.reexported = true;
+			} else {
+				variable.exportName = name;
+				variable.includeVariable();
+			}
+		});
+	}
+
+	link () {
```

Comment:
The name is a little vague as to what is actually linked to what. Maybe `linkDependencies`?